src/main/java/by/exadel/internship/service/impl/FileServiceImpl.java

```diff
@@ -0,0 +1,59 @@
+package by.exadel.internship.service.impl;
+
+import by.exadel.internship.exception_handing.FileNotUploadException;
+import by.exadel.internship.service.FileService;
+import com.google.auth.Credentials;
+import com.google.auth.oauth2.GoogleCredentials;
+import com.google.cloud.storage.*;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.net.URLEncoder;
+import java.nio.charset.StandardCharsets;
+
+import java.util.UUID;
+
+@Service
+public class FileServiceImpl implements FileService {
+
+    private static final String DOT_SEPARATOR = ".";
+
+    @Value("${firebase.url}")
+    private String downloadUrl;
+    @Value("${firebase.bucket}")
+    private  String bucketName;
+    @Value("${firebase.jsonFilePath}")
+    private String jsonFilePath;
+
+    @Override
+    public String upload(byte[] multipartFileInByte, String originalFileName) {
+        try {
```

Comment:
Please review this try block and make it beautiful!!!

Reply:
I hope i did it )

Reply:
Move log to the uploadFile method and replace try block with return .uploadFile(fileContent, fileName);

Reply:
Also you can move try-catch to uploadFile method as well

Reply:
Done